### DISCUSSION TITLE
Drawer closes if element inside clicked

### DISCUSSION THREAD

**Discussion by andrestrei** [chakra-ui/chakra-ui#7661] [Q&A]
I have a nextjs + react-leaflet + chakra setup. Got a button on the map that when clicked opens the drawer. In the drawer there's bunch of chakra switches user can enable/disable. Whenever I click on any of the switches, the drawer closes but I'd like it to stay open. Can't figure out why. Is there any prop on drawer component that controls this behaviour? Sandbox here [https://codesandbox.io/s/next-leaflet-chakra-drawer-x2txf9](url)

**Comment by souvik666** [comment]
hey not able to access your sandbox @andrestrei

**Reply by andrestrei** [reply]
weird, I have no problem opening it on different browsers, sometimes have to refresh though to make it work

**Reply by souvik666** [reply]
@andrestrei  showing this 
![image](https://github.com/chakra-ui/chakra-ui/assets/70321937/19345ec0-8041-4b93-8135-4181a47b42f8)

**Comment by HaleyWardo** [comment]
I am also unable to open the sandbox but have you tried preventing default on the toggle on change events? That's just my first thought.
